backend/capellacollab/projects/toolmodels/modelbadge/routes.py

```diff
@@ -41,13 +41,26 @@ async def get_model_complexity_badge(
     logger: logging.LoggerAdapter = fastapi.Depends(log.get_request_logger),
 ):
     try:
-        return responses.SVGResponse(
-            content=(
-                await git_handler.get_file_from_repository_or_artifacts(
-                    "model-complexity-badge.svg", "generate-model-badge"
-                )
-            )[1],
+        file = await git_handler.get_file("model-complexity-badge.svg")
```

Comment:
The previous `get_file_from_repository_or_artifacts` was really hard to understand because it implicitly assumed that you knew that we first look for a file on the revision given as argument and if that file is not found (i.e. an error is raised), we look for an artifact on the revision of the git handler itself. So I had to split them up to make this distinction clearer here, and because introducing caching would have made the function even more complex.
However, I agree that we can reintroduce it and use the two newly added functions. Anyway, I would keep the two functions, i.e. `get_file` and `get_artifact`, public in case you know you are only looking for a file or an artifact.